### ISSUE TITLE
analyzer/tool/summary/check_test fails on Windows

### ISSUE THREAD

**Issue by whesse** [dart-lang/sdk#25470]
I am marking it as Fail in the status file pkg/pkg.status.

Output of debug run on the buildbot:
stdout:
File: 'E:\b\build\slave\pkg-win-debug-be\build\sdk\pkg\analyzer\lib\src\summary\format.dart' does not have expected contents.
Please regenerate using:
  build\DebugIA32\dart-sdk\bin\dart.exe --package-root=file:///E:/b/build/slave/pkg-win-debug-be/build/sdk/build/DebugIA32/packages/ E:\b\build\slave\pkg-win-debug-be\build\sdk\pkg\analyzer\tool\summary\generate.dart

**Comment by whesse** [comment]
By the way, would the lines from check_test.dart that say:
  String script = Platform.script.toFilePath(windows: Platform.isWindows);
  String pkgPath = normalize(join(dirname(script), '..', '..'));

be better as
 String pkgPath = Platform.script.resolve('../..').toFilePath()

Resolve takes the filename off, attaches the string, and normalizes.
The default value of the windows: option is Platform.isWindows.

**Comment by stereotype441** [comment]
Example status file exhibiting the failure: https://build.chromium.org/p/client.dart/builders/pkg-win-be/builds/1495/steps/pkg%20tests/logs/stdio.  This build step tested commit f114dc1073d2f8705ad819d83267c8b6fe95588b.

I have been unable to reproduce the failure using a Windows laptop (I tried both commit f114dc1073d2f8705ad819d83267c8b6fe95588b and the latest commit, which is ba31abe42941c619ed98237254316a4bd843eec4), so I will have to debug this on the buildbot itself.

Unfortunately the buildbot is currently very unreliable (17 of the last 20 builds have failed to build the SDK, so they haven't even gotten to the point of running the test in question).  I've filed a bug to track that (#25474)--once it is fixed I will investigate why the buildbot fails this particular test.
